### PR TITLE
Implement considerations for Relax within osu!taiko diffcalc

### PR DESCRIPTION
**Pull request by Lawtrohux**
There's really not much to this. Implements considerations for relax. 

- Makes Relax incompatible with Swap and Random, as this is redundant with no colour in relax.
- Relax gives 0 rating to colour, a harsher scaling for converts, and a divided stamina difficulty due to the assumed increased finger count.

Shouldn't need a sheet or anything, as it is purely RX based.

### CODE REVIEW

File: osu.Game.Rulesets.Taiko/Mods/TaikoModRandom.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace osu.Game.Rulesets.Taiko.Mods
     public class TaikoModRandom : ModRandom, IApplicableToBeatmap
     {
         public override LocalisableString Description => @"Shuffle around the colours!";
-        public override Type[] IncompatibleMods => base.IncompatibleMods.Append(typeof(TaikoModSwap)).ToArray();
+        public override Type[] IncompatibleMods => base.IncompatibleMods.Append(typeof(TaikoModSwap)).Append(typeof(ModRelax)).ToArray();
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Taiko/Mods/TaikoModRandom.cs:18]: Are these changes necessary?

Comment (Lawtrohux) [reply osu.Game.Rulesets.Taiko/Mods/TaikoModRandom.cs:18]: Not necessary to relax diffcalc, just an afterthought.

Comment (Lawtrohux) [reply osu.Game.Rulesets.Taiko/Mods/TaikoModRandom.cs:18]: Copy that, and fair point. In the future if they are to be ranked, these randomisations could affect stamina values where they shouldn't for relax. I'm happy to leave these out for now as this is just a basic consideration.

Comment (Lawtrohux) [reply osu.Game.Rulesets.Taiko/Mods/TaikoModRandom.cs]: Changes reverted.

**Review by YaniFR — APPROVED** [review]

**Comment by Lawtrohux** [comment]
This will need to go in before the major pr, so i can merge conflict accordingly.

**Review by smoogipoo — APPROVED** [review]
